hs_tracking/views.py

```diff
@@ -24,6 +25,14 @@ def get(self, request, **kwargs):
 
         # log app launch details if user is logged in
         if request.user.is_authenticated():
+            res_id = querydict.pop('res_id', [''])[0]
+            tool_res_id = querydict.pop('tool_res_id', [''])[0]
+            if res_id and tool_res_id:
+                ret_exception = do_work_when_launching_app_as_needed(tool_res_id, res_id,
```

Comment:
yeah, I'm in favor of a try catch here instead of returning an exception string and checking it.

Reply:
@sblack-usu I think it makes sense in this case to have the web app utility function to handle exception and return exception message for the tracking view function to display to the user for better encapsulation of exception handling logic in the web app utility level. I am just trying to minimize the code needed to add to this tracking view function. Also, the web app view utility is supposed to handle exception specific to the needed work to be done when the web app is launched, e.g., in this case, when the user is not authenticated but the web app is set to be opened by a public resource, the geohub external web app being launched requires users to be authenticated via OAuth, so that is not really an exception, just a special case where a notification message needs to be displayed without proceeding to launch the app. Of course, the utility function can raise an exception in that special case for the tracking view to handle, but there may be more logic needed in the future to handle exceptions differently. So I think it is better to handle exception in the utility view function rather than leave exception to the more generic tracking view function. I will change the variable name to ```ret_exception_message``` to better reflect what it is, but I will leave the exception handling to the utility function as is. Let me know if you still have concerns on this.

Reply:
You can still handle exceptions within the do_work_when_launching_app_as_needed method as well as handle an exception here.  Just try/catch what you need, handle it and then raise an exception if a message needs to be passed up.  Then methods above it can catch those and parse the messages for user feedback.  I'm really just bringing up concern with this pattern I've seen of returning exception strings from a method, not limited to your work.  It's not an intuitive way to pass exception messages around.